pkg/ui/src/views/shared/components/dropdown/index.tsx

```diff
@@ -33,6 +34,29 @@ interface DropdownOwnProps {
  * Dropdown component that uses the URL query string for state.
  */
 export default class Dropdown extends React.Component<DropdownOwnProps, {}> {
+  dropdownRef: React.RefObject<HTMLDivElement>;
+  titleRef: React.RefObject<HTMLDivElement>;
+  selectRef: React.RefObject<ReactSelectClass>;
+
+  constructor(props: DropdownOwnProps) {
+    super(props);
+
+    this.dropdownRef = React.createRef();
+    this.titleRef = React.createRef();
+    this.selectRef = React.createRef();
+    this.triggerSelectClick = this.triggerSelectClick.bind(this);
+  }
+
+  triggerSelectClick(e: any) {
+    const dropdownNode: any = this.dropdownRef.current as Node;
+    const titleNode: any = this.titleRef.current as Node;
+    const selectNode: any = this.selectRef.current;
+
+    if (e.target === dropdownNode || e.target === titleNode || e.target.className.indexOf("dropdown__select") > -1) {
+      selectNode.handleMouseDownOnMenu(e);
```

Comment:
calling methods on a ref to a component instance is pretty wonky, but I don't know how else to make this work, and I don't know how much I care about the wonkiness…

Reply:
@vilterp thanks for the info re: commit/lint conventions.  I'll make the updates and resubmit.

Definitely agree the approach smells a little bit but I'm fairly certain something like this is what we're left with in absence of a means to pass a prop to the component that interacts with its internal state.  There is some grumbling about this in the repo so there might be an opportunity to refactor in the future if this ends up being merged.